src/main/java/Domain/Discounts/CategoryFixedDiscount.java

```diff
@@ -68,4 +69,9 @@ protected void applyDiscountLogic(ShoppingBasket basket) throws StockMarketExcep
             basket.setProductPriceToAmount(newpriceToAmount, product_id);
         }
     }
+
+    @Override
+    public BasicDiscountDto getDto() {
```

Comment:
why -1?

Reply:
category is not on specific product so id is not relevant